pkg/utils/functional/functional.go

```diff
@@ -74,3 +80,29 @@ func FilterMap[K comparable, V any](m map[K]V, f func(K, V) bool) map[K]V {
 	}
 	return ret
 }
+
+func GetExpirationTime(obj client.Object, provisioner *v1alpha5.Provisioner) time.Time {
```

Comment:
I think having it take in a `client.Object` is a reasonable middle state for now until we are able to pull out getting the node expiration time so that we don't repeat ourselves here.